src/pages/room/[id].tsx

```diff
@@ -1,142 +1,73 @@
-		:
-		room ? (
-			<Layout>
-				<div className="flex flex-wrap flex-auto justify-center md:space-x-32">
-					<div>
-						<SectionTitle hintColor="text-yellow-light-yellow">Players</SectionTitle>
-						{
-							room?.players.map((player: Player) => <UserProfile key={player.id} player={player} creatorId={room.creatorPlayerId}/>)
-						}
-						<Button onClick={copyLinkToClipboard}>
-							Copy link
-						</Button>
-					</div>
-					<div>
-						<Canvas width="800" height="400"
-						emitCoords={emitCoords}
-						broadcastedCoords={broadcastedCoords}
-						/>
-						<Title>Messages du salon :</Title>
-						<ul>
-						{messages.map((msg, key) => (
-							<li className="text-white-white" key={key}>
-							<span className="text-white-white font-bold">{msg.username}:</span> {msg.content}
-							</li>
-							))}
-							</ul>
-							<form action="" style={{display: 'flex'}}>
-							<input 
-							type="text" 
-							onChange={handleMessage} 
-							value={message}
-							className="bg-blue-200 w-full border-2 rounded border-yellow-light-yellow pl-2 text-white-white"/>
-							<Button onClick={sendMessage}>
-							Envoyer
-							</Button>
-							</form>
-						</div>
-						<div>
-							<SectionTitle hintColor="text-pink-dark-pink">Game</SectionTitle>
-						</div>
-				</div>
-			</Layout>
-		)
-		: (
-			<Layout>
-				<Title>Ce salon n`&apos;`existe pas, veuillez reessayer: </Title>
-				<Button><a href="/">Créer un salon / rejoindre un salon</a></Button>
-			</Layout> 
-		)
+    const getRoomId = (): string => {
+        if(typeof window !== "undefined") {
+            return RoomService.getRoomIdFromUrl(window.location.href)
+        }
+        return "";
+    }
+    
+    const roomId = getRoomId();
+    
+    const [isLoading, setIsLoading] = useState(true);
+    const [sessionId] = useLocalStorage("sessionId");
+    
+    const socket = useSocketRoom();
+    const [room, setRoom] = useState<RoomType>();
+    const [game, setGame] = useState<Game>();
+    
+    useEffect(() => {
+        if(!socket) return;		
+        socket.emit("join-room", roomId, (data) => {
+            if(data === false) {
+                // TODO: Redirect to homepage, because room does not exist or player is unable to join.
+            }
+            
+            setIsLoading(false)
+            setRoom(data);
+        })
+        
+        socket.on("update-room", (data) => {
+            setRoom(data);
+        })
+
+        socket.on("game-started", (room, game) => {
+            setRoom(room);
+            setGame(game);
+        })
+    }, [socket, sessionId, roomId]);
+    
+    return isLoading ? (
+        <Loading/>
+    )
+        :
+        room ? (
+            <RoomOrGame room={room} game={game}></RoomOrGame>
+        )
+        : (
+            <Layout>
+                <Title>Ce salon n`&apos;`existe pas, veuillez reessayer: </Title>
+                <Button><a href="/">Créer un salon / rejoindre un salon</a></Button>
```

Comment:
Ah ? Pourquoi tu escape comme ca ? C'est pas censé être nécessaire

Reply:
This is old code that was there much before I started working on the project. I'm letting it here for now because the lobby page will soon be remastered, and this will be removed (now useless since players will be redirected automatically)